package/MDAnalysis/core/groups.py

```diff
@@ -1923,6 +1996,128 @@ def __getattr__(self, attr):
+            selections ready to be applied to *base_group*.
+
+        """
+        # Because we're implementing __getattribute__, which needs _u for
+        # its check, no self.attribute access can be made before this line
+        self._u = base_group._u
+        self._selections = selections
+        self._base_group = base_group
+        self._lastupdate = None
+        self._derived_class = self._base_group._derived_class
+        if self._selections:
+            # Allows the creation of a cheap placeholder UpdatingAtomGroup
+            # by passing an empty selection tuple.
+            self.ensure_updated()
+
+    def update_selection(self):
+        """
+        Forces the reevaluation and application of the group's selection(s).
+
+        This method is triggered automatically when accessing attributes, if
+        the last update occurred under a different trajectory frame.
+
+        """
+        bg = self._base_group
+        sels = self._selections
+        if sels:
+            ix = sum([sel.apply(bg) for sel in sels])._ix
+        else:
+            ix = np.array([], dtype=np.int)
+        super(UpdatingAtomGroup, self).__init__(ix, bg._u)
+        self.is_uptodate = True
+
+    @property
+    def is_uptodate(self):
+        """
+        Checks whether the selection needs updating
+
+        Returns
+        -------
+        bool
+            `True` if the group's selection is up-to-date, `False` otherwise.
+
+        """
+        try:
+            return (object.__getattribute__(self, "_u").trajectory.frame ==
+                object.__getattribute__(self, "_lastupdate"))
+        except AttributeError: # self._u has no trajectory
+            return self._lastupdate == -1
+
+    @is_uptodate.setter
+    def is_uptodate(self, value):
+        value = bool(value)
+        if value:
+            try:
+                self._lastupdate = object.__getattribute__(self,
+                                                    "_u").trajectory.frame
+            except AttributeError: # self._u has no trajectory
+                self._lastupdate = -1
+        else:
+            # This always marks the selection as outdated
+            self._lastupdate = None
+
+    def ensure_updated(self):
+        """
+        Checks whether the selection needs updating and updates it, if needed.
+
+        Returns
+        -------
+        bool
+            `True` if the group was already up-to-date, `False` otherwise.
+
+        """
+        status = self.is_uptodate
+        if not status:
+            object.__getattribute__(self, "update_selection")()
+        return status
+
+    def __getattribute__(self, name):
+        if not (name in object.__getattribute__(self, "_shortcut_attrs")):
+            self.ensure_updated()
+        return object.__getattribute__(self, name)
+
+    def __repr__(self):
+        name = self.level.name
+        return ("<UpdatingAtomGroup with {} {}s>"
```

Comment:
Good idea. Right now only the parsed selections make it to the `UpdatingAtomGroup` but it won't be much trouble to pass the strings through as well; it's a lightweight extra.

The meaning of `__repr__` might become a bit misleading if `_base_group != u.atoms` but I think the user can live with it (or maybe I add a note to the text).